app/src/cpp/cfc-cpp/MultiThreadedDecoder.h

```diff
@@ -97,7 +96,8 @@ inline bool MultiThreadedDecoder::add(cv::Mat mat)
 		// if extracted image is small, we'll need to run some filters on it
 		clock_t begin = clock();
 		bool should_preprocess = (res == Extractor::NEEDS_SHARPEN);
-		unsigned decodeRes = _dec.decode_fountain(img, _writer, should_preprocess);
+		int color_correction = _legacyMode? 1 : 2;
```

Comment:
.

Reply:
Notably, this means that mode 4C will not benefit from the color correction matrix generated by mode B. I went back and forth on this, and probably will remove it once mode B is the default (if not sooner :thinking: ).

Mode B's CCM will pretty much always be better, *if* it exists.